### ISSUE TITLE
Regression in the command line parsing

### ISSUE THREAD

**Issue by dougwilson** [wycats/handlebars.js#1673]
I have been working to upgrade to the latest handlebars where I can (environment supports it), and I ran into an error and tracked it down to a regression in the handling of command line arguments when the same boolean was specified more than once.

For example, in version 4.7.3 and below, `handlebars --amd --no-amd` would result in the standard output, but in 4.7.4 and 4.7.5 it ends up resulting in AMD output, as if the `--no-amd` was not there.

The is appearing in some complex tooling internally where we have "default arguments" and some tools just add their own arguments for their own behavior. Think of it like `"handlebars --amd ${args} ${template}"` and the caller includes `--no-amd` in the args. This is a simplified example.

Here is the issue distilled down to the commands:

Version 4.7.5 (current):
```
$ touch empty
$ handlebars -v
4.7.5
$ handlebars --amd --no-amd empty
define(['handlebars.runtime'], function(Handlebars) {
  Handlebars = Handlebars["default"];  var template = Handlebars.template, templates = Handlebars.templates = Handlebars.templates || {};
return templates['empty'] = template({"compiler":[8,">= 4.3.0"],"main":function(container,depth0,helpers,partials,data) {
    return "";
},"useData":true});
});
```

Version 4.7.3:
```
$ touch empty
$ handlebars -v
4.7.3
$ handlebars --amd --no-amd empty
(function() {
  var template = Handlebars.template, templates = Handlebars.templates = Handlebars.templates || {};
templates['empty'] = template({"compiler":[8,">= 4.3.0"],"main":function(container,depth0,helpers,partials,data) {
    return "";
},"useData":true});
})();
```

Note that PR #1672 is a fix for this, liekly because it's just using the same parsing engine as all previous versions (minimist).

/cc @ErisDS

**Comment by ErisDS** [comment]
Sounds related to https://github.com/yargs/yargs/issues/465

TL;DR there's a parser option duplicate-arguments-array we can play with to see if we can get the original behaviour back.

**Comment by ErisDS** [comment]
Verified that the following resolves the regression: 

```
 .parserConfiguration({
    'duplicate-arguments-array': false
  })
```

**Comment by ErisDS** [comment]
@bcoe Oooooh if you can confirm it's fixed in 16 that's 👌 perfect. We've only got yargs on master now and only need support for Node v10 there :)

**Comment by dougwilson** [comment]
Yea, I was going to mention the same thing: master likely has no reason to not be on yargs 16, so if that works for everyone, then better all-around 🥂

**Comment by ErisDS** [comment]
Just tested and it works in 15 🙌 , bumping now 👍

**Comment by bcoe** [comment]
@ErisDS @dougwilson let me know if it still causes issues, and I can add a unit tests (I just seem to remember a fix around this, but might not be fully understanding).

**Comment by ErisDS** [comment]
Seems to be fixed to me. @dougwilson lemme know if you find any more issues like this 👍
